accepted/2021/install-location-per-architecture.md

```diff
@@ -138,34 +141,56 @@ any specific knowledge of where to find the install location, it would just
 invoke the right apphost.
 
 On all platforms the apphost will have a default location hardcoded
-and so if the installer puts the layout into that location it doesn't have to
+and so if the installer puts the product into that location it doesn't have to
 register that installation. But it would be cleaner if it did so anyway.
-Currently we always register the installation on Windows, but we don't seem
-to do so on Linux and macOS.
+Currently we always register the installation on Windows, but we don't do so
+on Linux and macOS.
 
-*Note: If installers write the file, they should also read it and validate
-that they don't overwrite it with different information. It's unclear yet
-if this is necessary and in which situations.*
+#### Windows installer behavior
 
-#### Support for downlevel apphost
+Pre-.NET 6 Windows installer writes registry keys for all architectures
+(even though it only installs one of them). Ideally the installer would
+only write the key for the architecture it's actually installing.
```

Comment:
This key is also only written by the SDK and not the runtime installers.  So if someone installs just runtimes the host won't have these keys.  Is that a problem?

Did you consider having the host install a key that you might use?  For instance adding InstallLocation here: https://github.com/dotnet/runtime/blob/2f68bf6a57890ac957943579c55a377265035b6d/src/installer/pkg/sfx/installers/host.wxs#L30-L32

Reply:
So if I install just the runtime, we don't write `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\InstallLocation`?

That would definitely be a problem - that is the one value which the host looks for - nothing else.

Reply:
Correct, it's only written by the SDK.  If you need to move we need to do that, or create a new key.  Moving can get tricky but it's might be possible.  Was discussing here https://github.com/dotnet/installer/issues/11999

Reply:
I thought that issue was about the fact that the installer writes keys for architectures it doesn't install.
Should I create new issue to track this (whether we fix it in 6 or not)?

Reply:
I mentioned this as a solution here: 
> If I could go back in time, I'd put this whole thing inside the MSIs. Host would register the "InstallLocation" and persist it with registry search on repair/upgrade. All the other MSI's would read it and honor it. They could block if the key wasn't present. Bundle would read it only for the purposes of blocking a move on upgrade, or a architecture clash. The only property the bundle would ever pass down would be "DOTNETHOME" in the case the user's choice made it past those blocks.

> It might be possible to do all this retroactively, but I'm not sure doing it in 6.0 RC2 is the best time.

Reply:
Per offline discussion we need to try to fix the MSI installers here.  I'll let you decide if you want to change the doc here at all.